crates/net/network/src/config.rs

```diff
@@ -81,6 +81,8 @@ pub struct NetworkConfig<C> {
     pub tx_gossip_disabled: bool,
     /// How to instantiate transactions manager.
     pub transactions_manager_config: TransactionsManagerConfig,
+    /// The NAT resolver for external IP
+    pub nat: Option<NatResolver>
```

Comment:
It may be worth calling this `external_ip_resolver` instead of `nat` for clarity